### ISSUE TITLE
Support uniform arrays in WebGL

### ISSUE THREAD

**Issue by emilio** [servo/servo#10440]
@Manishearth pointed out in https://github.com/servo/servo/pull/10436 that we're currently implementing `Uniformxyv` functions in terms of `Uniformxy`, which means that even when you're passing a vector of n elements we're just "filling" one of the uniform variables of the array.

For example, if you have in a vertex shader:

```glsl
uniform float myArray[2];
```

And you call:

```js
gl.uniform1fv(gl.getUniformLocation("myArray"), [2.0, 2.0]);
```

You'll expect `myArray[1]` in the shader to have the value `2`, while we'll only fill `myArray[0]`.

It's not a really used idiom but we should support it nontheless.

In order to support it, we should pass a `Vec<xxx>` to the backend (using `WebGLCommand`), check that the number of elements we're giving to it are a multiple of the size of the uniform.

Also it's a good option to review and make sure we're complaint with the spec in this functions.

**Comment by autrilla** [comment]
I'm working on this, by the way, so assign it if you want to

**Comment by autrilla** [comment]
Fixed by #10436

**Comment by KiChjang** [comment]
@autrilla On your next PR, if your commit message (or PR description) contains the words "fix/fixes/address/addresses/close/closes #(issue no.)", then GitHub will automatically close the relevant issue once the PR lands. Thanks!
